ui/desktop/src/goosed.ts

```diff
@@ -59,8 +59,46 @@ const checkServerStatus = async (
   return false;
 };
 
+const connectToExternalBackend = async (
+  workingDir: string,
+  port: number = 3000
+): Promise<[number, string, ChildProcess]> => {
+  log.info(`Using external goosed backend on port ${port}`);
+
+  const isReady = await checkServerStatus(port);
+  if (!isReady) {
+    throw new Error(`External goosed server not accessible on port ${port}`);
+  }
+
+  if (process.platform === 'win32') {
+    throw new Error('External backend process discovery not supported on Windows');
+  }
+
+  let pid: number;
+  try {
+    const stdout = execSync(`lsof -ti:${port}`, { encoding: 'utf8' });
+    const parsedPid = parseInt(stdout.trim());
+    if (isNaN(parsedPid)) {
+      throw new Error('Could not parse PID from lsof output');
+    }
+    pid = parsedPid;
+  } catch (error) {
+    throw new Error(`Could not find process for external backend on port ${port}: ${error}`);
+  }
+
+  const mockProcess = {
+    pid,
```

Comment:
do we actually need the pid of the backend if we aren't managing the process? maybe it isn't worth unwinding but it would let you skip the whole lsof thing

Reply:
done